### PR TITLE
E2E CI test postgres data backend

### PR DESCRIPTION
**Pull request by ewgenius**
Added test flow to test postgres data backend https://docs.spiceai.org/connectors/postgres/

- starts postgres
- generates example table
- installs spice and set up spice app, with postgres datasets connection

Most recent test run - https://github.com/spiceai/spiceai/actions/runs/8370542233

### CODE REVIEW

File: .github/workflows/e2e_test_ci.yml [hunk 1]
```diff
@@ -5,15 +5,18 @@ on:
...
+  pull_request:
+    branches:
+      - trunk
 
   workflow_dispatch:
 
 jobs:
   build:
+    # WIP
+    if: false
```
Comment (ewgenius) [inline .github/workflows/e2e_test_ci.yml]: TODO: enable back

File: .github/workflows/e2e_test_ci.yml [hunk 2]
```diff
@@ -5,10 +5,10 @@ on:
...
       - trunk
       - release-*
-  
-  # pull_request:
-  #   branches:
-  #     - trunk
+
+  pull_request:
+    branches:
+      - trunk
```
Comment (ewgenius) [inline .github/workflows/e2e_test_ci.yml]: TODO:

remove

File: .github/workflows/e2e_test_ci.yml [hunk 3]
```diff
@@ -352,4 +356,157 @@ jobs:
           if [[ $sql_output == *"error"* ]] || [[ $sql_output == *"not found"* ]]; then
             echo "Unexpected response from spice sql, failed to perform test query: $sql_output"
             exit 1
-          fi
\ No newline at end of file
+          fi
+
+  test_quickstart_data_postgres:
+    name: "E2E Test: SpiceAI quickstart, using ${{ matrix.target_os }}-${{ matrix.target_arch }}"
+    runs-on: ${{ matrix.runner }}
```
Comment (sgrebnov) [inline .github/workflows/e2e_test_ci.yml:363]: SpiceAI quickstart (Postgres) or SpiceAI Postgres quickstart

**Review by sgrebnov — DISMISSED** [review]

File: .github/workflows/e2e_test_ci.yml [hunk 4]
```diff
@@ -352,4 +356,157 @@ jobs:
           if [[ $sql_output == *"error"* ]] || [[ $sql_output == *"not found"* ]]; then
             echo "Unexpected response from spice sql, failed to perform test query: $sql_output"
             exit 1
-          fi
\ No newline at end of file
+          fi
+
+  test_quickstart_data_postgres:
+    name: "E2E Test: SpiceAI quickstart, using ${{ matrix.target_os }}-${{ matrix.target_arch }}"
```
Comment (ewgenius) [inline .github/workflows/e2e_test_ci.yml]:
```suggestion
    name: "E2E Test: SpiceAI Postgres quickstart, using ${{ matrix.target_os }}-${{ matrix.target_arch }}"
```

**Review by sgrebnov — APPROVED** [review]

**Pull request merged**